kube/kubeclient/client.go

```diff
@@ -257,7 +257,7 @@ type JSONPatch struct {
 // It currently (2023-03-02) only supports "add" and "remove" operations.
 func (c *client) JSONPatchSecret(ctx context.Context, name string, patch []JSONPatch) error {
 	for _, p := range patch {
-		if p.Op != "remove" && p.Op != "add" {
+		if p.Op != "remove" && p.Op != "add" && p.Op != "replace" {
 			panic(fmt.Errorf("unsupported JSON patch operation: %q", p.Op))
```

Comment:
👍🏼 will add a TODO to potentially remove. I didn't add this bit, so it might be either that it was added in error or there is some reason I can't think of now

Reply:
Actually I really can't see why we would need to panic, so returning an error now

Reply:
Thanks for spotting this